extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.moat-tracking.js

```diff
@@ -2,17 +2,36 @@ define('wikia.articleVideo.featuredVideo.moatTracking', [
 	'ext.wikia.adEngine.adContext',
 	'wikia.window'
 ], function (adContext, win) {
-	return function(player) {
+
+	var jwplayerPluginUrl = 'https://z.moatads.com/jwplayerplugin0938452/moatplugin.js',
+		partnerCode = 'wikiajwint101173217941';
+
+	function track(player) {
 		if (adContext.get('opts.isMoatTrackingForFeaturedVideoEnabled')) {
 			player.on('adImpression', function (event) {
 				if (win.moatjw) {
 					win.moatjw.add({
 						adImpressionEvent: event,
-						partnerCode: 'wikiajwint101173217941',
+						partnerCode: partnerCode,
 						player: this
 					});
 				}
 			});
 		}
-	};
+	}
+
+	function loadTrackingPlugin() {
+		if (!win.moatjw) {
+			var scriptElement = win.document.createElement('script');
+			scriptElement.async = true;
+			scriptElement.src = jwplayerPluginUrl;
+			win.document.head.appendChild(scriptElement);
+		}
+	}
+
+	track.track = track;
```

Comment:
I personally don't like this trick. Assigning attributes to a function makes it much harder to understand what's going on here.